component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java

```diff
@@ -109,91 +159,202 @@ void unregisterSourceListener(String listenerUrl) {
+            this.listenerConfiguration = listenerConfiguration;
+        }
+
+        public void incrementReferenceCount() {
+            this.referenceCount++;
+        }
+
+        public void decrementReferenceCount() {
+            this.referenceCount--;
+        }
+
+        public ServerConnector getServerConnector() {
+            return this.serverConnector;
+        }
+
+        public ListenerConfiguration getListenerConfiguration() {
+            return this.listenerConfiguration;
+        }
+
+        public int getReferenceCount() {
+            return this.referenceCount;
+        }
+    }
+
+    /**
+     * This method wil check that if there is already registered server connectors which may be http but if it have
+     * jks security setup then it can be use as https transport as well
+     * listener configuration
+     *
+     * @param listenerConfiguration server listener configuration.
+     * @param context               server connector context handler
+     * @return conflict exits or not.
+     */
+    private boolean checkForConflicts(ListenerConfiguration listenerConfiguration,
+                                      HttpServerConnectorContext context) {
+        if (context == null) {
+            return false;
+        }
+        if (listenerConfiguration.getScheme().equalsIgnoreCase("https")) {
+            ListenerConfiguration config = context.getListenerConfiguration();
+            if (config.getScheme().equalsIgnoreCase("https")) {
+                if (!listenerConfiguration.getKeyStoreFile().equals(config.getKeyStoreFile())
+                        || !listenerConfiguration.getKeyStorePass().equals(config.getKeyStorePass())
+                        || !listenerConfiguration.getCertPass().equals(config.getCertPass())) {
+                    log.info("There is already registered https server connector for same host:port which has " +
+                            " conflicting configurations.");
+                    return true;
                 }
+            } else {
+                return true;
             }
         }
+        return false;
+    }
+
+    private void setConnectorListeners(ServerConnectorFuture connectorFuture, String serverConnectorId,
+                                       ConnectorStartupSynchronizer startupSyncer) {
+        connectorFuture.setHttpConnectorListener(new HTTPConnectorListener());
+        connectorFuture.setPortBindingEventListener(
+                new HttpConnectorPortBindingListener(startupSyncer, serverConnectorId));
+    }
+
+    private void validateConnectorStartup(ConnectorStartupSynchronizer startupSyncer) {
+        int noOfExceptions = startupSyncer.getExceptions().size();
+        if (noOfExceptions <= 0) {
+            return;
+        }
+        PrintStream console = System.err;
+
+        startupSyncer.getExceptions().forEach((connectorId, e) -> {
+            console.println("siddhi: " + e.getMessage() + ": [" + connectorId + "]");
```

Comment:
1. Isn't logging initialized at this point?
2. Shall we also add the cause to the error message